### PR TITLE
Preparations and refactoring

### PR DESCRIPTION
**Pull request by lethal-guitar**
Contains some of the changes from #429. Since these are valuable improvements to the code, I'd like to merge these changes sooner, as #429 still needs a bit more work.

### CODE REVIEW

**Comment by lethal-guitar** [comment]
@pratikone this one is ready for review now

File: src/renderer/renderer.cpp [hunk 1]
```diff
@@ -339,6 +339,14 @@ data::Image createWaterSurfaceAnimImage() {
     static_cast<size_t>(WATER_MASK_HEIGHT * WATER_NUM_MASKS)};
 }
 
+
+auto getSize(SDL_Window* pWindow) {
```
Comment (pratikone) [inline src/renderer/renderer.cpp]: just curious on the lambda being converted to a function. It seems to be called only from one place.

File: src/main.cpp [hunk 2]
```diff
@@ -14,20 +14,23 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+// Welcome to the RigelEngine code base! If you are looking for the place in
```
[the remark this answers is not in the document]
Comment (lethal-guitar) [reply src/main.cpp]: Thanks, I plan to add many more of this type!

File: src/renderer/renderer.cpp (hunk 1 above)
Comment (lethal-guitar) [reply src/renderer/renderer.cpp]: Ah yeah, it's indeed used only once on this branch - this is a preparation for some changes on #429, where we will need it in more than one place. I'll add a note to the commit message to clarify that!

File: src/data/game_options.hpp [hunk 3]
```diff
@@ -19,10 +19,31 @@
...
 constexpr auto SOUND_VOLUME_DEFAULT = 1.0f;
 
+
+/** Data-model for user-configurable options/settings
+ *
+ * This struct contains everything that can be configured by the user in
+ * RigelEngine. The corresponding UI is located in ui/options_menu.cpp
+ * Serialization code is found in common/user_profile.cpp
+ *
+ * If you add something to this struct, you most likely want to add
```
[the remark this answers is not in the document]
Comment (lethal-guitar) [reply src/data/game_options.hpp]: Good point, that would be nice indeed! I'm not sure if there's a reliable way to assert it, but I'll have a look. Adding comments to the other places is definitely a good idea in any case.

**Comment by lethal-guitar** [comment]
@pratikone thanks for your review!

File: src/data/game_options.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (lethal-guitar) [reply src/data/game_options.hpp]: I need to look into more options for asserting. For now, I've added some comments.

**Pull request merged**